copy/multiple.go

```diff
@@ -140,6 +218,32 @@ func (c *copier) copyMultipleImages(ctx context.Context) (copiedManifest []byte,
 				UpdateSize:                  int64(len(updated.manifest)),
 				UpdateCompressionAlgorithms: updated.compressionAlgorithms,
 				UpdateMediaType:             updated.manifestMIMEType})
+		case instanceCopyClone:
+			logrus.Debugf("Replicating instance %s (%d/%d)", instance.sourceDigest, i+1, len(instanceCopyList))
+			c.Printf("Replicating image %s (%d/%d)\n", instance.sourceDigest, i+1, len(instanceCopyList))
+			unparsedInstance := image.UnparsedInstance(c.rawSource, &instanceCopyList[i].sourceDigest)
+			if instance.compressionVariant != nil {
+				// ---- THIS BLOCK IS A HACK and will not be merged till: https://github.com/containers/image/pull/2050
+				// copy options with new compression format
+				if c.options.DestinationCtx == nil {
+					c.options.DestinationCtx = &types.SystemContext{}
+				}
+				c.options.DestinationCtx.CompressionFormat = instance.compressionVariant
+			}
+			updated, err := c.copySingleImage(ctx, unparsedInstance, &instanceCopyList[i].sourceDigest, true)
```

Comment:
(BTW it might make sense for `prepareInstanceCopies` to prepare full `copySingleImage` values, instead of carrying the data one field at a time in `instanceCopy`.)

Reply:
…` copySingleImageOptions` that is